### ISSUE TITLE
not compatible with inplace operation?

### ISSUE THREAD

**Issue by chihchiehchen** [Araachie/koala#1]
Hello,

Thanks for the inspiring work. I'm very interested in your work and try to implement koala for segmentation tasks. I begin with vanilla U-Net, unfortunately I got an error message below:

> one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [6, 64, 1, 1]] is at version 3; expected version 1 instead.

 If possible, could you give me some hint to resolve this problem?

Thanks,
Chih-Chieh

**Comment by Araachie** [comment]
Dear Chih-Chieh,

I highly appreciate and thank you for your interest in our work.

To my experience, such error indicates that the parameters of your network are changed between the forward and the backward pass. This can happen, when, for example, you use multiple losses and call backward after the optimization step.
It would be nice, if you could share your training code. This would help resolve the issue and see if there is something wrong with our code.

Feel free to contact me directly via email.

Best,
Aram

**Comment by chihchiehchen** [comment]
Hello,

Sorry I'm late. You are right, I put optimizer in the wrong place ( before model(input)). Now everything works well. Thanks a lot!

Best,
Chih-Chieh
